pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -102,10 +103,27 @@ func RunStartBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args
 		return cmdutil.UsageError(cmd, "Must pass a name of a BuildConfig or specify build name with '--from-build' flag")
 	}
 
-	name := buildName
-	isBuild := true
-	if len(name) == 0 {
-		name = args[0]
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	var (
+		name        = buildName
+		kind        = ""
+		defaultKind = "BuildConfig"
+		isBuild     = true
+	)
+
+	if len(name) == 0 && len(args[0]) > 0 {
+		mapper, _ := f.Object()
+		kind, name, err = osutil.ResolveResource(defaultKind, args[0], mapper)
+		if err != nil {
+			return err
+		}
+		if kind != defaultKind {
```

Comment:
> or if we don't want to do that, we should handle the "builds" case specifically by telling them to use --from-build=..., rather than "invalid resource provided"

That's a good suggestion